File: spring-boot/src/main/java/com/bp/experiment/dao/model/Usergroupstate.java
package com.bp.experiment.dao.model;

import org.hibernate.annotations.ColumnDefault;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "usergroupstate")
public class Usergroupstate {
    @Id
    @ColumnDefault("nextval('usergroupstate_id_seq')")
    @Column(name = "id", nullable = false)
    private Integer id;

    @NotNull
    @Column(name = "label", nullable = false, length = Integer.MAX_VALUE)
    private String label;

}

File: gin/models/usergroupstate.go
package models

type Usergroupstate struct {
	ID    int    `gorm:"column:id;primaryKey;autoIncrement;default:nextval('usergroupstate_id_seq')"`
	Label string `gorm:"column:label;not null"`
}

func (Usergroupstate) TableName() string {
	return "usergroupstate"
}
